packages/fxa-payments-server/src/components/PaymentForm/index.tsx

```diff
@@ -208,18 +234,17 @@ export const PaymentForm = ({
 
       {confirm && plan && (
         <Localized
-          id="payment-confirm"
-          $interval={plan.interval}
+          id={`payment-confirm-${plan.interval}`}
+          $intervalCount={plan.interval_count}
           $amount={formatCurrencyInCents(plan.amount)}
```

Comment:
I looked at `formatCurrencyInCents` in `lib/formats.ts`, and I'd like to flag it as an issue from the localization POV.

```ts
export function formatCurrencyInCents(amount: number) {
  return (amount / 100.0).toFixed(2);
}
```

The function converts the number to a string using English rules of number formatting, i.e. using the dot `.` as the decimal separation. This won't work for many languages out there.

The right approach is to use [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat/NumberFormat) which can format numbers using language specific rules. However, rather than use it here (which would also require access to the current `locale`), you can rely on Fluent doing the right thing for you :)

Fluent auto-formats all numbers passed into translations, but it uses `Intl.NumberFormat` default options which won't work for currencies. Options can be customized by developers in the source code, and then also overriden by localizers in the translations.

#### Custom options set in the source code

These effectively become the new defaults for the one particular number their set for and will be used in all translations, unless explicitly overriden by the localizer. Rather than pass `$amount={formatCurrencyInCents(plan.amount)}` you can instantiate a `FluentNumber`:

```ts
$amount={new FluentNumber(plan.amount, {
    style: "currency",
    currency: "USD",
    currencyDisplay: "symbol",
})}
```

With the above definition of `$amount`, you don't need the `$` symbol in front of the amount in the translation:

```properties
## subscription upgrade plan details
plan-price-day = {$intervalCount ->
    [one] ${$amount} daily
   *[other] ${$amount} every {$intervalCount} days
}
```

↑ becomes ↓

```properties
## subscription upgrade plan details
plan-price-day = {$intervalCount ->
    [one] {$amount} daily
   *[other] {$amount} every {$intervalCount} days
}
```

#### Per-locale option overrides

Localizers can then customize number formatting options if they think it would produce a better translations. For instance, they could choose to change `currencyDisplay` from `symbol` (`5,01 $` in Polish, note the comma) to `code` to better emphasize the fact that this is a foreign currency (`5,01 USD` in Polish).

```properties
## subscription upgrade plan details
plan-price-day = {$intervalCount ->
    [one] {$amount} daily
   *[other] {NUMBER($amount, currencyDisplay: "code")} every {$intervalCount} days
}
```

#### The Caveat

There's one caveat with this approach, which you should be aware of. Currently, all options can be overriden by localizer, including the `currency` itself! This obviously shouldn't be possible. I've meant to fix this for a long time, but there never was a use-case which would make it important, until right now. I'll be happy to fix it in `@fluent/bundle`.

Reply:
@stasm That sounds great. I am wondering if it might be a good idea to create a separate issue for the purpose of currency display? The reason for this is that right now, the front-end is assuming `USD` in all cases (This is because we only offer subscriptions to US-based customers), however, we can update the API to provide the currency alongside the amount, which would allow for a more future-proof solution, combined with the update to `@fluent/bundle`.
So, what I would like to propose: create a new branch that has an updated version of `@fluent/bundle` and our subscription API, so that we can use localized currency with confidence.

Reply:
I'm all for the iterative approach, and I'll be happy to help with the upgrade to `@fluent/bundle` PR.

One thing to keep in mind: depending on when you expose these strings to localizers, you'll need to change their identifiers if you make a change to the `$amout` variable. This way Pontoon will let localizers know that they need to update their translations.

It might be a good idea to remove that `$` symbol right now and for example prepend it in `formatCurrencyToCents`, temporarily.

Reply:
This temporary change has been made